From: Tomas Henzl <thenzl@redhat.com>
Date: Sun, 29 Aug 2010 15:49:33 -0400
Subject: [block] cciss: fix memory leak in cciss_init_one
Message-id: <1283097002-3341-35-git-send-email-thenzl@redhat.com>
Patchwork-id: 27880
O-Subject: [RHEL6 PATCH 34/63] cciss: fix memory leak in cciss_init_one
Bugzilla: 568830
RH-Acked-by: Neil Horman <nhorman@redhat.com>

fix memory leak in cciss_init_one

diff --git a/drivers/block/cciss.c b/drivers/block/cciss.c
index 087c951..d2793c1 100644
--- a/drivers/block/cciss.c
+++ b/drivers/block/cciss.c
@@ -4151,7 +4151,7 @@ static int __devinit cciss_init_one(struct pci_dev *pdev,
 	int j = 0;
 	int rc;
 	int dac, return_code;
-	InquiryData_struct *inq_buff = NULL;
+	InquiryData_struct *inq_buff;
 
 	if (reset_devices) {
 		/* Reset the controller with a PCI power-cycle */
@@ -4288,6 +4288,7 @@ static int __devinit cciss_init_one(struct pci_dev *pdev,
 		printk(KERN_WARNING "cciss: unable to determine firmware"
 			" version of controller\n");
 	}
+	kfree(inq_buff);
 
 	cciss_procinit(i);
 	hba[i]->cciss_sector_size = 2048;
